util/profile/profile.go

```diff
@@ -3,13 +3,28 @@ package profile
 import (
 	"net/http"
 	"net/http/pprof"
+	"os"
+
+	"github.com/argoproj/argo-cd/v2/util/env"
 )
 
+var enableProfilerFilePath = env.StringFromEnv("ARGOCD_ENABLE_PROFILER_FILE_PATH", "/home/argocd/.enable-profiler")
+
+func wrapHandler(handler http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if _, err := os.Stat(enableProfilerFilePath); err == nil {
+			handler.ServeHTTP(w, r)
+		} else {
+			http.NotFound(w, r)
```

Comment:
A 404 might be confusing, as it's actually not configured, which isn't a client error, but a server error. There isn't quite a fit of HTTP status codes that match this though. A 5xx might be appropriate here, or I've also unironically previously suggested a 418 as an indication of mismatching config.

Also printing directions in the response could be useful to users.

But that can all be adjusted post-hoc in a future PR as well.

Reply:
Great suggestion about printing the directions - updated response. 
Agree about 404 - it might be difficult to understand if the URL is wrong or the feature is disabled. IMHO 5xx might be misleading as well if server actually cannot handle the response. I've changed response code to 401.

Reply:
Yea existing HTTP status codes don't really have a good fit for it (other than 418, which is an easter egg and would also be confusing to use as an actual code 😅 ). 401 makes enough sense to me, but I'll let others consider if that's the most optimal response (as there is no authorization here either, just a configuration)